src/v/serde/rw/scalar.h

```diff
@@ -0,0 +1,92 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#pragma once
+
+#include "bytes/iobuf_parser.h"
+#include "serde/rw/tags.h"
+#include "serde/serde_exception.h"
+#include "serde/serde_is_enum.h"
+#include "serde/type_str.h"
+#include "ssx/sformat.h"
+#include "vlog.h"
+
+#include <cinttypes>
+#include <type_traits>
+
+namespace serde {
+
+template<typename To, typename From>
+To bit_cast(From const& f) {
```

Comment:
Can we use `std::bit_cast` instead?

Reply:
True. This code was written when we could not use it yet.